livre/core/pipeline/AsyncData.h

```diff
@@ -0,0 +1,113 @@
+/* Copyright (c) 2011-2016, EPFL/Blue Brain Project
+ *                     Ahmet Bilgili <ahmet.bilgili@epfl.ch>
+ *
+ * This file is part of Livre <https://github.com/BlueBrain/Livre>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3.0 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#ifndef _AsyncData_h_
+#define _AsyncData_h_
+
+#include <livre/core/types.h>
+
+namespace livre
+{
+
+/**
+ * The AsyncData class provides thread safe operations for setting/retrieving/querying
+ * the data. The internal data has a name and data type.
+ */
+class AsyncData
+{
+public:
+
+    /**
+     * When connection is instantiated the data is not
+     * set yet, so any get() call will block the retrieval
+     * @param name of the connection
+     * @param dataType type of the data
+     */
+    explicit AsyncData( const DataInfo& dataInfo );
```

Comment:
What's the advantage of taking a single parameter instead of the string and data type. I know that in C++11 there's no much difference because you can invoke the constructor like this: AsyncData({ name, type }), but its less explicit.